cmd/sonobuoy/app/pluginList.go

```diff
@@ -171,3 +200,9 @@ func loadManifest(bytes []byte) (*manifest.Manifest, error) {
 	err := kuberuntime.DecodeInto(manifest.Decoder, bytes, &def)
 	return &def, errors.Wrap(err, "couldn't decode yaml for plugin definition")
 }
+
+// isNotExist returns true if the cause of the error was that the plugin did not exist.
+func isNotExist(e error) bool {
```

Comment:
The doc for `pluginNotFoundErr` says
```
pluginNotFoundError is a custom type so we can tell whether or not loading the plugin from the installation directory FAILED or if it just wasn't found.
```
Could this not be done with Go io errors ?

Reply:
> Go io errors

What do you mean by that? You usually either have an exact error you check against (e.g. `if err==notFoundErr`) or a type which you can typecast and check. The only other complication is with wrapped errors but that doesn't really change those two approaches.

Reply:
Oh I see what not found means, it has nothing to do with actual file op, but a look up in the cache.  Never mind.

```go
func loadPlugin(installedDir, reqFile string) (*manifest.Manifest, error) {
	// Ignore error since it may not have to do with this plugin at all. Only return error if not found.
	// The loadPlugins will log errors as needed for visibility anyways.
	manMap, _ := loadPlugins(installedDir)
	reqManifest := manMap[filepath.Join(installedDir, reqFile)]
	if reqManifest != nil {
		return reqManifest, nil
	}
	return nil, &pluginNotFoundError{fmt.Errorf("failed to find plugin file %v within directory %v", reqFile, installedDir)}
}
```